examples/platforms/efr32mg12/alarm.c

```diff
@@ -137,15 +137,24 @@ void otPlatAlarmMilliStartAt(otInstance *aInstance, uint32_t t0, uint32_t dt)
     expires_microsec = (t0 + dt) * US_IN_MS;
     status           = RAIL_SetTimer(gRailHandle, expires_microsec, RAIL_TIME_ABSOLUTE, RAILCb_TimerExpired);
 
-    if (status == RAIL_STATUS_NO_ERROR)
+    if (status != RAIL_STATUS_NO_ERROR)
     {
-        sIsRunning = true;
-    }
-    else
-    {
-        sIsRunning = false;
-        otLogCritPlat("Alarm set timer, status: %d, dt: %u, t0: %u", status, dt, t0);
+        // The RAIL timer could not be set due to expiration time being in the past with respect to RAIL's current
+        // time which is in microseconds. We fallback to using a relative timer from the current time.
+
+        expires_microsec = dt * US_IN_MS;
+        status           = RAIL_SetTimer(gRailHandle, expires_microsec, RAIL_TIME_DELAY, RAILCb_TimerExpired);
+
+        if (status != RAIL_STATUS_NO_ERROR)
+        {
+            uint32_t now = otPlatAlarmMilliGetNow();
```

Comment:
nit: can we put the call to `otPlatAlarmMilliGetNow()` in the argument list for `otLogCritPlat()`?

Reply:
Added the recommended change.